packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart

```diff
@@ -158,29 +157,23 @@ class StreamVideoPushNotificationManager implements PushNotificationManager {
     });
 
     _subscriptions.add(
-      _idCallKitIncoming,
-      onCallEvent.whereType<ActionCallIncoming>().listen(
-        (_) {
-          if (!client.isConnected) {
-            client.openConnection();
-          }
-
-          subscribeToEvents();
-        },
-      ),
-    );
-
-    _subscriptions.add(
-      _idCallKitAcceptDecline,
-      onCallEvent.whereType<ActionCallAccept>().map((_) => null).mergeWith([
-        onCallEvent.whereType<ActionCallDecline>().map((_) => null),
-        onCallEvent.whereType<ActionCallTimeout>().map((_) => null),
-      ]).listen(
+      _idCallKit,
+      onCallEvent.listen(
         (event) {
-          _subscriptions.cancel(_idCallAccepted);
-          _subscriptions.cancel(_idCallEnded);
-          _subscriptions.cancel(_idCallRejected);
-          _subscriptions.cancel(_idCallParticipantCount);
+          if (event is ActionCallIncoming) {
+            if (!client.isConnected) {
+              client.openConnection();
+            }
+
+            subscribeToEvents();
+          } else if (event is ActionCallAccept ||
+              event is ActionCallDecline ||
+              event is ActionCallEnded) {
```

Comment:
Is replacing `ActionCallTimeout` with `ActionCallEnded` intentional?

Reply:
nice catch!! Both should be there - fixed